src/mpid/ch4/netmod/ofi/ofi_rma.c

```diff
@@ -47,6 +47,8 @@ void MPIDI_OFI_complete_chunks(MPIDI_OFI_win_request_t * winreq)
     }
 
     winreq->chunks = NULL;
+    MPIR_Datatype_release_if_not_builtin(winreq->noncontig.get.origin.datatype);
```

Comment:
This needs a check of the RMA request type so it can access the datatype fields using the right union type.